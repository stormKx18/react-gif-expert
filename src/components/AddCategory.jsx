import { useState } from 'react'

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({target}) =>{
        setInputValue(target.value)
    }

    const onSubmit = (event)=>{
        event.preventDefault();
        const cleanInput = inputValue.trim()
        if(cleanInput.length<=1) return;
        //setCategories((categories)=>[inputValue,...categories]);
        onNewCategory(cleanInput)
        setInputValue('')
    }

  return (
    <form onSubmit={onSubmit} aria-label='form'>
        <input 
        type="text" 
        placeholder="Search gifs..."
        value={inputValue}
        onChange={onInputChange}
        />
    </form>
  )
}

