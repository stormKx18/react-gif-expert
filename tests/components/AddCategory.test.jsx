import { render, screen, fireEvent } from "@testing-library/react"
import { AddCategory } from "../../src/components"

describe('Tests for <AddCategory />', () => {

  test('Component should match snapshot ', () => {
    const {container} = render(<AddCategory onNewCategory={()=>{}}/>);
    expect(container).toMatchSnapshot();
  })

  test('Component should change the value of the text box', () => {
    const inputValue = 'Saitama';
    render(<AddCategory onNewCategory={()=>{}}/>);

    const inputBox = screen.getByRole('textbox');
    fireEvent.input(inputBox,{target:{value:inputValue}});
    expect(inputBox.value).toBe(inputValue);
    // screen.debug()
  })
  
  test('Component should call onNewCategory if the input has a value', () => {
    const inputValue    = 'Saitama';
    const onNewCategory = jest.fn();

    render(<AddCategory onNewCategory={onNewCategory}/>);

    const inputBox = screen.getByRole('textbox');
    const form = screen.getByRole('form');
    fireEvent.input(inputBox,{target:{value:inputValue}});
    fireEvent.submit(form);
    
    expect(inputBox.value).toBe('');
    expect(onNewCategory).toHaveBeenCalled();
    expect(onNewCategory).toHaveBeenCalledTimes(1);
    expect(onNewCategory).toHaveBeenCalledWith(inputValue);
    // screen.debug()
  })
  
  test('Component should not call onNewCategory if the input is empty', () => {
    const inputValue    = '';
    const onNewCategory = jest.fn()

    render(<AddCategory onNewCategory={onNewCategory}/>);
    const inputBox = screen.getByRole('textbox');
    const form = screen.getByRole('form');

    fireEvent.input(inputBox,{target:{value:inputValue}});
    fireEvent.submit(form);

    expect(onNewCategory).not.toHaveBeenCalled();

  })
  
  
})
