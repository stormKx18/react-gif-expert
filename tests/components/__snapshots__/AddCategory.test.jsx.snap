// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <AddCategory /> Component should match snapshot  1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs..."
      type="text"
      value=""
    />
  </form>
</div>
`;
